src/bin/cargo/cli.rs

```diff
@@ -387,6 +390,14 @@ fn config_configure(
     Ok(())
 }
 
+/// Precedence isn't the most obvious from this function because
+/// - Some is determined by `expand_aliases`
+/// - Some is enforced by `avoid_ambiguity_between_builtins_and_manifest_commands`
+///
+/// In actuality, it is:
+/// 1. built-ins xor manifest-command
+/// 2. aliases
```

Comment:
We may have something like below but won't work 🥲. So, no global args for `cargo <file>`?

```bash
cargo foo.rs --config 'alias."foo.rs" = bar.rs --offline --verbose'
```

The other point is that `cargo foo.rs -Zscript` doesn't work. Is it intentional? I know eventually `-Zscript` won't be there but people may run `cargo foo.rs -Zno-index-update` and bang!

Reply:
All args to `cargo` must come between `cargo` and `foo.rs`.

Specifically calling `cargo foo.rs --config 'alias."foo.rs" = bar.rs --offline --verbose'` won't work because `foo.rs` will be a "third party subcommand" and all arguments after it get captured.  We then expand aliases and append those extra arguments at the end of the alias.  This means the alias expansion will never see the `--config` call, making setting aliases after `foo.rs` meaningless.

Reply:
What doesn't work for `cargo --config "alias…" foo.rs`?

Reply:
I was a fool. Aliases have lower precedence.